packages/headless/src/utils/validate-payload.test.ts

```diff
@@ -57,7 +57,7 @@ describe('validatePayloadAndThrow', () => {
 });
 
 describe('validateOptions', () => {
-  let engine: Engine<unknown>;
+  let engine: Engine<object>;
```

Comment:
Was this a necessary change, or just a nice to have/nice improvement ?

If it's a necessary change, this could point to a potential breaking change in the API.

Reply:
Indeed, I'd say it's more of a improvement than necessary. Strictly speaking, it is breaking.

The first generic arg is the shape of the state. The state must be an object (it cannot be a string, array etc.)
I adjusted the generic to reflect this:

```
// before
Engine<State = SearchAppState>

// after
Engine<State extends object = SearchAppState>
```

I needed to do this because I placed the same restriction on `CoreEngine`, to force the restriction on all `State` generics extending the interface.

I will relax the restriction to be safe and add it for v1.